tests/test_gates.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the correct application of gates with lightning.qubit
+"""
+import itertools
+
+import numpy as np
+import pennylane as qml
+import pytest
+
+from pennylane_lightning import LightningQubit
+
+
+@pytest.fixture
+def op(op_name):
+    return getattr(qml, op_name)
+
+
+@pytest.mark.parametrize("op_name", LightningQubit.kernel_operations)
```

Comment:
That's a good point! From a quick look through those tests, I agree that they provide similar coverage to what's here. I'm generally happy to have some redundancy because:
- This is directly in `lightning.qubit` and it is easier for developers to quickly see this failing (often, at least for me, running the dev test suite is more of an afterthought)
- I didn't see this exact test in the suite, so maybe there is some edge case covered (not sure)!

Really, having it here explicitly makes me feel more confident that things are definitely working :laughing:

Reply:
Agreed 🤔 Calling the device test suite can be overlooked. Could we
1) Call on the device test suite from the lightning tests (not quite sure about the best way to do this 🤔 )
2) extend `make test` with the device test command?

For 1), we could maybe call `pl-device-test` from one of the test files 🤔 

My worry is that we'd start off adding test cases here which will essentially diverge from the device test suite and we'll have to maintain both.

Reply:
Yes I agree maintainability is important! Because I'm paranoid about double making sure that things work correctly now for `lightning.qubit`, I still prefer to potentially have some repetition with a benefit of increased confidence. If down the line we see issues with these local tests, we could delete them (and potentially update the dev test suite accordingly).